Added session argument to Navigator.hal factory method.

tspng opened the pull request:
That's my first pull request, hope it works.

I also updated the authentication part of the documentation, hope that's ok like that (English is not my mother tongue).

Commit:
Added session argument to Navigator.hal factory method.

bubenkoff commented:
please add small test for this change

tspng commented:
Sorry, completely forgot about the tests.
Class TestCreate in test_hal_nav2py would be the correct place, right?

bubenkoff commented:
i think just a test_session(
a function in that module

On 30 April 2015 at 12:09, haestan notifications@github.com wrote:

> Sorry, completely forgot about the tests.
> Class TestCreate in test_hal_nav2py would be the correct place, right?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/deontologician/restnavigator/pull/38#issuecomment-97727602
> .

## 

Anatoly Bubenkov

tspng commented:
Right, TestCreate is about the create() method, my bad.

deontologician commented on restnavigator/halnav.py:
```diff
@@ -167,7 +167,13 @@ class Navigator(object):
+            default_curie=None,
+            auth=None, 
+            headers=None, 
+            session=None
```
could you add a trailing comma after this last line? if we add more keyword args in the future, we won't have to modify it

deontologician commented:
This is looking good! I like that you added an example, that's awesome

deontologician commented on README.rst:
```diff
@@ -262,6 +262,18 @@ Now we can put it all together to create a new post:
+It is also possible to specify a custom requests Session object when creating 
+a new navigator.
+
+For example, if you want to talk to a OAuth2 protected api, simply pass 
```
Should be "an OAuth2", otherwise the grammar is perfect

Commit:
Added a test for custom Navigator's session argument

tspng commented:
Ok, fixed the comma and spelling issue.
Added a test case for the session argument, but it's a really stupid test. Just comparing if the passed session argument ends up at _core's session property.

deontologician commented:
Thanks!
